src/github.com/matrix-org/dendrite/mediaapi/writers/download.go

```diff
@@ -101,3 +118,152 @@ func (r *downloadRequest) Validate() *util.JSONResponse {
 	}
 	return nil
 }
+
+func (r *downloadRequest) doDownload(w http.ResponseWriter, cfg *config.MediaAPI, db *storage.Database, activeRemoteRequests *types.ActiveRemoteRequests) *util.JSONResponse {
+	// check if we have a record of the media in our database
+	mediaMetadata, err := r.getMediaMetadata(db)
+	if err == nil {
```

Comment:
There was a reason for doing is this way, but it's been reworked as part of addressing the `sql.ErrNoRows` 'leak'. Done.